frontend/src/scenes/retention/queries.ts

```diff
@@ -0,0 +1,51 @@
+import { RetentionTableAppearanceType, RetentionTablePeoplePayload } from 'scenes/retention/types'
+
+import { query } from '~/queries/query'
+import { NodeKind, PersonsQuery, RetentionQuery } from '~/queries/schema'
+
+export function retentionToActorsQuery(query: RetentionQuery, selectedInterval: number, offset = 0): PersonsQuery {
+    const group = query.aggregation_group_type_index !== undefined
+    const select = group ? 'group' : 'person'
+    return {
+        kind: NodeKind.PersonsQuery,
+        select: [select, 'appearances'],
+        orderBy: ['length(appearances) DESC', 'actor_id'],
+        source: {
+            kind: NodeKind.InsightPersonsQuery,
+            interval: selectedInterval,
+            source: {
+                ...query,
+                retentionFilter: {
+                    ...query.retentionFilter,
+                },
+            },
+        },
+        offset,
+        limit: offset ? offset * 2 : undefined,
+    }
+}
+
+function appearances_1s_0s(appearances: number[], totalIntervals: number, selectedInterval: number | null): number[] {
+    const newTotalIntervals = totalIntervals - (selectedInterval ?? 0)
+    return Array.from({ length: newTotalIntervals }, (_, intervalNumber) =>
+        appearances.includes(intervalNumber) ? 1 : 0
+    )
+}
+
+export async function queryForActors(
+    retentionQuery: RetentionQuery,
+    selectedInterval: number,
+    offset: number = 0
+): Promise<RetentionTablePeoplePayload> {
+    const actorsQuery = retentionToActorsQuery(retentionQuery, selectedInterval, offset)
+    const response = await query(actorsQuery)
+    const results: RetentionTableAppearanceType[] = response.results.map((row) => ({
+        person: row[0],
+        appearances: appearances_1s_0s(row[1], retentionQuery.retentionFilter.total_intervals || 11, selectedInterval),
```

Comment:
Could this be done at the SQL level? 🤔 
I think it'd be nicer and easier for the users, if they get a list/array/sequence of 0-s and 1-s  back directly from the query. That makes further analysis easier (`appearances[1] == true`)

Reply:
I somehow found it easier to understand that I get a list of intervals: I ask for week, total_intervals = 10, and selected interval 2 and get back [2,3,4]. A list of 1s and 0s or even booleans would IMO be harder to manage for someone wanting to use the data somewhere else.